### PR TITLE
split-key handlers implementation in wasm

### PR DESCRIPTION
**Pull request by Ash-KODES**
### Changes
-Added methods of Split key in wasm required for web flow implementation in bolt web app.

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [x]  Merge your system tests PR to master AFTER merging this PR

### CODE REVIEW

File: wasmsdk/wallet_base.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"encoding/json"
+	"strings"
+	"github.com/0chain/errors"
+	"github.com/0chain/gosdk/core/zcncrypto"
+)
+
+var _config localConfig
```
Comment (peterlimg) [inline wasmsdk/wallet_base.go]: The `_config` is defined in zcncore package, what we need to do is just call the `SplitKeys` and `SetWalletInfo`.

File: wasmsdk/wallet_base.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	StatusSuccess      int = 0
+	StatusNetworkError int = 1
+	// TODO: Change to specific error
+	StatusError            int = 2
+	StatusRejectedByUser   int = 3
+	StatusInvalidSignature int = 4
+	StatusAuthError        int = 5
+	StatusAuthVerifyFailed int = 6
+	StatusAuthTimeout      int = 7
+	StatusUnknown          int = -1
```
Comment (peterlimg) [inline wasmsdk/wallet_base.go]: Let's use them via `zcncore` package instead of re-define them here.

File: wasmsdk/wallet_base.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	}
+	w, err := sigScheme.SplitKeys(numSplits)
+	if err != nil {
+		return "", errors.Wrap(err, "split key failed.")
+	}
+	wStr, err := w.Marshal()
+	if err != nil {
+		return "", errors.Wrap(err, "wallet encoding failed.")
+	}
+	return wStr, nil
```
Comment (peterlimg) [inline wasmsdk/wallet_base.go]: call `zcncore.SplitKeys(privateKey, numSplits)`

File: wasmsdk/wallet_base.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+//  - splitKeyWallet: if wallet keys is split
+func setWalletInfo(jsonWallet string, splitKeyWallet bool) error {
+	err := json.Unmarshal([]byte(jsonWallet), &_config.wallet)
+	if err == nil {
+		if _config.chain.SignatureScheme == "bls0chain" {
+			_config.isSplitWallet = splitKeyWallet
+		}
+		_config.isValidWallet = true
+	}
+	return err
```
Comment (peterlimg) [inline wasmsdk/wallet_base.go:24]: Call `zcncore.SetWalletInfo()`

File: wasmsdk/wallet_base.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+// - url: the url of zAuth server
+func setAuthUrl(url string) error {
+	if !_config.isSplitWallet {
+		return errors.New("", "wallet type is not split key")
+	}
+	if url == "" {
+		return errors.New("", "invalid auth url")
+	}
+	_config.authUrl = strings.TrimRight(url, "/")
+	return nil
```
Comment (peterlimg) [inline wasmsdk/wallet_base.go]: Call `zcncore.SetAuthUrl()`

File: wasmsdk/transaction_base.go [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package main
+
+import "github.com/0chain/gosdk/core/zcncrypto"
+
+type localConfig struct {
```
Comment (peterlimg) [inline wasmsdk/transaction_base.go]: not needed as they are in `zcncore` package

File: wasmsdk/transaction_base.go [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+	chain         ChainConfig
+	wallet        zcncrypto.Wallet
+	authUrl       string
+	isConfigured  bool
+	isValidWallet bool
+	isSplitWallet bool
+}
+
+type ChainConfig struct {
```
Comment (peterlimg) [inline wasmsdk/transaction_base.go]: Same as above

**Review by peterlimg — DISMISSED** [review]

**Review by dabasov — APPROVED** [review]
lgtm

**Pull request merged**